api/src/test/java/org/apache/iceberg/types/TestTypes.java

```diff
@@ -79,6 +95,77 @@ public void fromPrimitiveString() {
     assertThatExceptionOfType(IllegalArgumentException.class)
         .isThrownBy(() -> Types.fromPrimitiveString("abcdefghij"))
         .withMessage("Cannot parse type string to primitive: abcdefghij");
+
+    assertThat(Types.fromPrimitiveString("geometry")).isEqualTo(Types.GeometryType.crs84());
+    assertThat(Types.fromPrimitiveString("Geometry")).isEqualTo(Types.GeometryType.crs84());
+    assertThat(Types.fromPrimitiveString("geometry(srid:3857)"))
+        .isEqualTo(Types.GeometryType.of("srid:3857"));
+    assertThat(Types.fromPrimitiveString("geometry( srid:3857 )"))
+        .isEqualTo(Types.GeometryType.of("srid:3857"));
+    assertThat(Types.fromPrimitiveString("geometry( srid: 3857 )"))
+        .isEqualTo(Types.GeometryType.of("srid: 3857"));
+    assertThat(Types.fromPrimitiveString("Geometry( projjson:TestIdentifier )"))
+        .isEqualTo(Types.GeometryType.of("projjson:TestIdentifier"));
+
+    assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(() -> Types.fromPrimitiveString("geometry()"))
+        .withMessageContaining("Cannot parse type string to primitive");
```

Comment:
@szehon-ho what do you think the behavior here should be? I would expect it to match the following case and throw `Invalid CRS: (empty string)`.

Reply:
Yea I think so, the current message is not that clear.

Reply:
I've tweaked the regex to match this regex with the first group being empty, now it will emit `Invalid CRS: (empty string)`.